signac/contrib/job.py

```diff
@@ -569,6 +574,8 @@ def _check_manifest(self):
         manifest = self._read_manifest()
         if calc_id(manifest) != self.id:
             raise JobsCorruptedError([self.id])
+        if self._statepoint is None:
+            self._statepoint = SyncedAttrDict(manifest, parent=_sp_save_hook(self))
```

Comment:
This is an optimization: if we have to load and validate the manifest, we might as well set the job's lazy-loaded state point here if it hasn't been set yet.

Reply:
👍